doc/explanation/package-management.md

```diff
@@ -0,0 +1,226 @@
+# How Package Management Works
+
+This document gives an explanation on how the new package management
+feature introduced in Dune works under the hood. It requires a bit of
+familiarity with how opam repositories work and how Dune builds packages. Thus
+it is aimed at people who want to understand how the feature works, not how it
+is used.
+
+For a tour on how to apply package management to a project, refer to the
+{doc}`/tutorials/dune-package-management/index` tutorial.
+
+## Motivation
+
+A core part of modern programming is using existing code to save time.
+The OCaml package ecosystem has quite a long history with many projects
+building upon each other over many years. A significant step forward was the
+creation of the OCaml Package Manager, opam, along with the establishment of a
+public package repository which made it a lot more feasible to share code
+between people and projects.
+
+Over time, best practices have evolved, and while opam has incorporated some
+changes, it couldn't adopt all the modern workflows due to its existing user base and constraints. 
+
+Dune Package Management attempts to take the parts of the opam ecosystem that
+have stood the test of time and couple them with a modern workflow. Some of the
+improvements include:
+
+* Automatic package repository updates
+* Easily reproducible dependencies
+* All package dependencies declared in a single file that is kept in-sync
+* Per-project dependencies
+
+Dune plays well with the existing OCaml ecosystem and does not introduce a new
+type of packages. Rather, it uses the same package repository and Dune packages stay
+installable with opam.
+
+## Package Management in a Project
+
+This section describes what happens in a Dune project using the package
+management feature.
+
+## Dependency Selection
+
+The first step is to determine which packages need to be installed.
+Traditionally this has been defined in the `depends` field of a projects opam
+file(s).
+
+Since a while Dune has also supported {doc}`opam file generation
+</howto/opam-file-generation>` by specifying the package dependencies in the
+`dune-project`. Outside of this feature, Dune had not used the `depends` stanza.
```

Comment:
There's something off with the beginning of this sentence. Do you mean: "For a while, Dune has also supported..." or "Since Dune has supported...."

Reply:
Thanks for pointing it out. The "a while" was my attempt to weasel out of the fact that I don't know since when this existed. This pushed me to open #10984 so I can write "1.10" here and it makes more sense.